boot/initramfs.go

```diff
@@ -0,0 +1,72 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package boot
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/snap"
+)
+
+// TODO:UC20: add install/recover mode mounts code here too?
+
+// InitramfsRunModeChooseSnapsToMount returns a map of the snap paths to mount
+// for the specified snap types.
+func InitramfsRunModeChooseSnapsToMount(
+	typs []snap.Type,
+	blDir string,
+	blOpts *bootloader.Options,
```

Comment:
do we strictly need the bootloader.Options ?

Reply:
hmm, I guess I need to check if there will be a bootloader on /boot/grub and whether we will auto-find the right one in `/EFI/ubuntu` without the option set, but I'm pretty sure we do need this

Reply:
sorry, I mean to be need them passed in? or we basically know what we need (like makebootable) ?

Reply:
As I suppose with decisions we made we don't need blDir either?

Reply:
we need modeenv because it's used both here and by the caller

Reply:
ah I see what you mean, yes I suppose we could specify those just in this function instead of in snap-bootstrap